src/app/mypage/[id]/page.tsx

```diff
@@ -0,0 +1,46 @@
+// src/app/my-page/[id]/page.tsx
+'use client'
+
+import React, { useEffect, useState } from 'react';
+import { useRouter } from 'next/navigation';
+import { getAuth, onAuthStateChanged } from "firebase/auth";
+import { auth } from '@/lib/firebase/config';
+
+const MyPage = () => {
+  const [loading, setLoading] = useState(true);
+  const [user, setUser] = useState(null);
```

Comment:
このStateに型指定がないです。いまだとnullしかStateに入れられないようになってます。
```suggestion
  const [user, setUser] = useState<User | null>(null);
```

Reply:
typescriptのエラーが表示されてると思うので修正しておくといいと思います！